### PR TITLE
Move to

### PR DESCRIPTION
**Pull request by lisarics**

### CODE REVIEW

File: app/models/chess_piece.rb [hunk 1]
```diff
@@ -2,6 +2,34 @@ class ChessPiece < ApplicationRecord
...
+          update_attributes(self.position_x = stop[0], self.position_y = stop[1])
+          capture_piece!(board, stop)
+    else 
+      raise ArgumentError.new('you cannot capture your own player')
+    end
+  end
+ 
+  # DESTROYS A PIECE ON THE BOARD
+  def capture_piece!(board, stop)
+    board[stop[1]][stop[0]] == nil
```
Comment (jakekarl3) [inline app/models/chess_piece.rb:19]: I think this should be a single '=' to change the value on the board. == will compare and return a true/false

File: app/models/chess_piece.rb [hunk 2]
```diff
@@ -2,6 +2,34 @@ class ChessPiece < ApplicationRecord
 
   belongs_to :game
 
+  # UPDATES POSITION IF IT'S EMPTY OR CAN BE CAPTURED OR RAISES ERROR IF IS OWN PLAYER
+  def move_to!(stop)
```
Comment (jakekarl3) [inline app/models/chess_piece.rb:6]: Pretty sure the logic works, however I think we're doing some of this twice. The valid_move method for king, queen, and bishop checks if the desired move is valid by checking if the space is occupied, is it the opponent, and if the path is obstructed. Not sure how we want to go about adjusting either of those.

File: spec/models/game_spec.rb [hunk 3]
```diff
@@ -2,6 +2,7 @@
```
Comment (jakekarl3) [inline spec/models/game_spec.rb:2]: Just a general comment to remove the excess white space

**Review by jakekarl3 — COMMENTED** [review]
I think we have some of the checks for a valid move doubled up in the move_to! and valid_move methods

File: app/models/chess_piece.rb (hunk 2 above)
Comment (lisarics) [reply app/models/chess_piece.rb:6]: Not sure what you're referring to here.  I didn't recreate those methods just moved them up the page.

Comment (lisarics) [reply app/models/chess_piece.rb:6]: according to the ticket the move_to! method is supposed to check if there's an occupying piece and if it's the opposite color and then update the attributes.  It's in the process of getting hooked up in the controller. so it's basically the capture logic.

**Pull request closed without merging**